File: megumi/src/main/java/com/erisu/cloud/megumi/command/Command.java
package com.erisu.cloud.megumi.command;

import com.erisu.cloud.megumi.pattern.Pattern;
import com.erisu.cloud.megumi.util.Module;

import java.lang.annotation.*;

/**
 * 指令系统基本配置
 */

@Target({ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface Command {

    CommandType commandType();  //  群聊....

    Pattern pattern();

    String value() default "";    //匹配值

    String[] alias() default {};

//    boolean isSpec() default false;// 是否特殊化处理

    String uuid() default "";

    /**
     * 概率回复函数
     *
     * @return
     */
    double probaility() default 1.0; //没想好怎么实现更优雅

}

File: megumi/src/main/java/com/erisu/cloud/megumi/command/Command.kt
package com.erisu.cloud.megumi.command

import com.erisu.cloud.megumi.pattern.Pattern

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class Command(
    val commandType: CommandType, //  群聊....
    val pattern: Pattern, val value: String = "", //匹配值
    val alias: Array<String> = [],  //    boolean isSpec() default false;// 是否特殊化处理
    val uuid: String = "",
    val probaility: Double = 1.0, //概率随机函数

    val isRecalled: Boolean = false,//是否撤回消息
    val recallTime: Long = 0,// 撤回时间:s
    val prefix: String = "",//  前缀
)

